### DISCUSSION TITLE
2.0.0 Beta started with Solarbank 2 and Smart Meter support

### DISCUSSION THREAD

**Discussion by thomluther** [thomluther/ha-anker-solix#93] [Announcements]
## 2.0.0 Pre-Release
Enhancements have been added to the Api library and the HA integration to **provide support for Solarbank 2 and Smart Meter devices.**
The power usage mode of the SB2 should be changeable with this initial release, however any schedule modifications for SB2 are not supported at this time. This may be supported in a future release after appropriate helper methods are developed to support modification options with the new SB2 schedule format.
**This release comes with a breaking change for the energy statistic entities.** Read them carefully in the [2.0.0 pre-release notes](https://github.com/thomluther/hacs-anker-solix/releases/tag/2.0.0).
The enhancements are included in the updated [INFO.md](https://github.com/thomluther/hacs-anker-solix/blob/main/INFO.md)
I have upgraded to this pre-release as well with my Solarbank 1 setup and haven't found any issues yet.
But this is no measure for the new Solarbank 2 systems and Smart Meters, which must be tested by the community.
A big thank you for all users on the [Api Library repo](https://github.com/thomluther/anker-solix-api) who contributed with configuration exports and Api testing!. Without your assistance, these enhancements would not have been possible.
But there is still work to do. If you want to contribute and explore the Api and the existing or new endpoints with your system and your Anker main account, please see the [open issues](https://github.com/thomluther/anker-solix-api/issues) and assist in exploration, testing and documentation where applicable.
Since I own only a Solarbank 1 System, I cannot explore any new cloud Api capabilities on my own, therefore further enhancements will remain limited and just be possible with assistance by the community.

## How to install beta and pre-releases with HACS easily
When you enabled experimental features in your HACS configuration, HACS will provide update entities for each of the HACS installed integrations. See the last item in the HACS configuration options.
![image](https://github.com/user-attachments/assets/d9de24cb-2eeb-415f-bbff-e3eacbc6afdc)
Then you can simply use the HA core 'install update service' which allows you to specify a labeled version for the update of the selected entity. The version will first be looked up in the github repo tags (releases or pre-releases) and if not found there it will use the branch tags. So you can upgrade and downgrade your integration to any existing (pre-) release or branch of the repo. Likewise you can also install the 'main' branch if that contains the latest code but has not been tagged with a release yet.

To install this pre-release, just select the Anker-Solix update entity and specify the 2.0.0 version
![image](https://github.com/user-attachments/assets/e5cb07ab-c6fc-4dec-b91b-861f1933dc7e)

After installation, you need to restart home assistant. But before restarting, copy the [optional update of the entity images](https://github.com/thomluther/hacs-anker-solix#optional-entity-pictures) to your www folder as described in the README.
If you find any problems or weird reporting, please enable debug mode of the integration, recreate the problem and open a new issue with the debug logs.
There is also a new Get system info service, where you can check the actual data response from the cloud Api on your own.

## Attention:
Since I cannot test the new version with the new devices I will leave this version active for a couple of days before releasing it as new version. Fixes or further improvements may require changes in the [Api library](https://github.com/thomluther/anker-solix-api) too.

**Comment by thomluther** [comment]
Please provide feedback if all is working well. In particular:
- Can the usage mode be changed between manual and automatic mode when the smart meter is installed
- Without installed smart meter, is only the manual usage mode available in the select list?
- Are the energy statistics all correct as shown in the App? The energy stats also work for shared accounts, so they could be compared real time. Each energy sensor has the previous day in the attributes, so you may also check the last full day for comparing the totals.

Note: Anker changed something on the cloud on 25.07. With that change, the cloud no longer returns invalid data to ensure proper data display on home screens for shared accounts. That means you should not longer see unavailable sensors in the integration, even if you do NOT skip invalid responses (since there are no longer invalid responses).
However, the data update interval was not changed. It is still 5 minutes when not triggered for frequent updates by the open Home Screen in the Anker app. You should be able to see that in the sensor or cloud data timestamp history, which will change probably only every 5 minutes.

Another change that was done in the cloud is the response for automatic firmware updates. Previously this returned the SB2 at least as a configurable device. Now the response field is null. Typically I would expect an empty List if there is no configurable device. So it is not clear whether this is a cloud bug that was introduced and therefore also not showing this option anymore in the App, or whether SB2 firmware really was changed to no longer provide auto-update capability. 
Anyway, since this response field is now null, no auto-update entities will be created for SB2 systems.

**Comment by kimi1983** [comment]
Hello,

I am using your plugin since 2.0.0 beta, and I am happy but sometimes HA is not updating values until i reopen the mobile app on my phone.
But I am still not sure, what kind of user should be used in HA ? The main account or a secondary one ?

best regards,
Kim

**Reply by Moshkopp** [reply]
normally use Primary Acc in HA, because of changing data ist not possible with the second one

**Reply by thomluther** [reply]
Hi, that's the way SB2 works as [described in the INFO](https://github.com/thomluther/hacs-anker-solix/blob/main/INFO.md#option-considerations-for-solarbank-2-devices-and-smart-meters)
It sends data updates to the cloud only every 5 minutes, unless you open the App with the main account. There is nothing the Api can do to change this behavior.

You can use the shared account if you do not want to change anything via the integration, otherwise the main account may be better, but then you have to disable the Api usage switch when you want to log into your App again while avoiding to get kicked out.

**Reply by kimi1983** [reply]
I am not sure to understand what you mean by : disable the Api usage switch when you want to log into your App

But ok by design the better way would be to use the main account in HA ? In this case HA will automatically update 60 seconds ?

**Reply by thomluther** [reply]
No, the update frequency does not change with the type of account you use in HA.
But when you use an account in HA, you cannot use the same in the App.
If you use the main account in HA, and want to login to the App at the same time with same account, you have to disable the Api usage switch in your HA Anker system switch entity, otherwise you will be kicked out in the App when the integration runs the next refresh with the same account.
You can use an automation that simplifies the temporary account usage in the App and automatically reactivate it in the Integration once done with the App. See [INFO](https://github.com/thomluther/hacs-anker-solix/blob/main/INFO.md#automation-to-send-and-clear-sticky-actionable-notifications-to-your-smart-phone-based-on-api-switch-setting)

**Reply by thomluther** [reply]
If you do not want to change any of the settings, use a shared account in the HA integration

**Reply by kimi1983** [reply]
Ok but with the shared account the data update is not as fast as with the main account ?

What is the "Api Nutzung" Switch for ?
In case of using the main account for HA I will gett quicker updates ?

sorry,

**Reply by kimi1983** [reply]
in my case i could use the main account for HA, and the shared account for my smartphone, right ?
Will HA be updated as fast as possible ?

**Reply by thomluther** [reply]
I thought I explained that all before. The account type does NOT make difference how fast the data update.
This is fix at 5 minute intervals, unless you open the Anker App with your main account all the time!

**Comment by kimi1983** [comment]
Hm I am not sure about that.
Will stop opening the app with my main account right now at 18.00. I will check if Ha still updates without manual App refresh. Will keep you updated.

**Reply by thomluther** [reply]
it will update with the defined interval of your configuration (every 60 seconds per default), BUT the values of the power entities will not change for 5 minutes since they will not be updated in the cloud. You can check the SB data time entity in the system which gives you an idea when the data was sent to the cloud the last time. However this timestamp is not always reliable, since if it is invalid from the cloud, it is replaced with the actual time when the request response was received by the integration

**Comment by kimi1983** [comment]
![Screenshot_20240729-181046](https://github.com/user-attachments/assets/d4181b28-213c-45f5-91a4-3d96895209f2)
Have a look, last update of Ha was 18 minutes ago.. since i stopped opening my mobile app around 17.55.

In my case, ha running with shared account is not updating at all...

**Comment by kimi1983** [comment]
**
![image](https://github.com/user-attachments/assets/d06679ac-876e-4d5d-9009-7662fa671002)
**

Now i switched HA to my main account, SB Datenzeit keeps updating but no data is changing. (2 hours delay for SB Datenzeit)

Am I the only one with this problem ?

**Reply by thomluther** [reply]
I also wonder why your Total Yield is unknown and the other totals like CO2 and Cost Savings are 0 as well?
Did you just install the system?
If not, why is there no data collection in the cloud for those totals?
What is your app showing for the totals?

**Reply by kimi1983** [reply]
Since my installation 10 days ago I am still waiting for the statistics. Same for the mobile Anker App, no Statistics...
Maybe my update problem is related to this.

**Reply by neTix84** [reply]
Its a known issue, for the most people it works to delete the battery and add it again.

**Reply by thomluther** [reply]
Maybe, you should make sure that there is no timezone difference in your setup.
Maybe you have to recreate the whole power system in the app, because something is weird with it. Otherwise the cloud would count the totals and also the energy statistics...
And you should NOT see a 2 hour delay in the SB Datenzeit timestamp.

**Reply by kimi1983** [reply]
Ok will do a reset of the whole system.
Will keep you updated.

best regards,
Kim

**Reply by kimi1983** [reply]
Added a second battery. Deleted the Solarbabk 2, and delete the System.

Created a new System with a new Name and added Solarbank and Smartmeter to it.

How fast should I see stats in my App?

**Reply by neTix84** [reply]
In my case, it took minutes. 

Otherwise i would recommend to contact Anker directly.

**Reply by kimi1983** [reply]
Hm ok will try to see with Anker.
- deleted my SB2
- Deleted my system
- Deleted my anker account
Recreated everything from scratch but still no stats...

**Reply by thomluther** [reply]
Did you plug off all cables from the SB2 and did a total reset of the system pressing the button for 7 seconds?
Some users reported just after such a total reset, the system behaved normal, recognized the battery SOC and allowed firmware upgrades to complete. Maybe that will resolve your time zone offset and also the data update problem with the cloud too

**Reply by thomluther** [reply]
Maybe you should read through well PV forum posts for such type of problems, because Anker support is slow.
I can recommend the German Photovoltaik Forum for such technical discussions.
This[ user also fixed his SB2 with a total reset and new setup from scratch](https://www.photovoltaikforum.com/thread/228451-anker-solix-2-die-neue-solarbank/?postID=3867561#post3867561)

**Reply by kimi1983** [reply]
Since this morning stats are visible in my anker App :) 
But I think the only thing which helped was a full reset by holding both buttons for 9 seconds.

Thank you for your help.

**Reply by neTix84** [reply]
Good that you found a Solution 👍🏻

**Comment by kimi1983** [comment]
![image](https://github.com/user-attachments/assets/dd70b2fe-a94e-42b5-bd25-ed31f1aa9c16)
 as you can see on the right SB Datenzeit is the only value which is updating.

**Comment by thomluther** [comment]
For the 2 hour time difference, I assume there is still a mismatch in timezone settings between your mobile App and your HA host.
I don't see a gap of 2 hours, but 2 hours is the difference between CET summer time and GMT.

If you want to see what data is in the cloud, you can also use the [service Get System Info](https://github.com/thomluther/hacs-anker-solix/blob/main/INFO.md#get-system-info-service).
This response will be used to update most of the entities.

**Reply by kimi1983** [reply]
Even after reset of my system i am still 2 hours behind the SB Datenzeit.
My time settings in HA are correct, since all other sensors and integrations are using the right time.

**Reply by thomluther** [reply]
I cannot comment on any of those questions, I have no SB2

**Reply by kimi1983** [reply]
Since my Anker App is now reporting historical data I did a new config of my system in HA.
This time with my main account, but SB Datenzeit is still 2 hours behind,,

Here you can find an "service Get System Info" in which we can see historical data ;

sensor.system_mini_pv_ertrag_gesamt:
  system_info:
    home_info:
      home_name: Home
      home_img: ""
      charging_power: "0.00"
      power_unit: W
    solar_list: []
    pps_info:
      pps_list: []
      total_charging_power: "0.00"
      power_unit: W
      total_battery_power: "0.00"
      updated_time: ""
      pps_status: 0
    statistics:
      - type: "1"
        total: "0.49"
        unit: kWh
      - type: "2"
        total: "0.49"
        unit: kg
      - type: "3"
        total: "0.10"
        unit: €
    topology_type: "1"
    solarbank_info:
      solarbank_list:
        - device_pn: A17C1
          device_sn: APCGQ80E22200075
          device_name: Solarbank 2 E1600 Pro
          device_img: >-
            https://public-aiot-fra-prod.s3.dualstack.eu-central-1.amazonaws.com/anker-power/public/product/2024/05/24/iot-admin/5iJoq1dk63i47HuR/picl_A17C1_normal%281%29.png
          battery_power: "16"
          bind_site_status: ""
          charging_power: "365"
          power_unit: W
          charging_status: "0"
          status: "1"
          wireless_type: "1"
          main_version: ""
          photovoltaic_power: "791"
          output_power: "365"
          create_time: 1722373844
          set_load_power: ""
          sub_package_num: 1
          output_cutoff_data: 10
          is_display: true
      total_charging_power: "426"
      power_unit: W
      charging_status: "0"
      total_battery_power: "0.16"
      updated_time: "1970-01-01 00:00:00"
      total_photovoltaic_power: "791"
      total_output_power: "365.00"
      display_set_power: false
      battery_discharge_power: "0"
      ac_power: "0"
      to_home_load: "365"
      is_display_data: true
      solar_power_1: "197"
      solar_power_2: "200"
      solar_power_3: "199"
      solar_power_4: "195"
    retain_load: 0W
    scene_mode: 1
    home_load_power: "365"
    updated_time: 01-01-0001 00:00:00
    power_site_type: 5
    site_id: c8eccfab-f02a-4c09-bec7-76adc9a3776f
    powerpanel_list: []
    grid_info:
      grid_list:
        - device_pn: A17X7
          device_sn: AUJS020E17412691
          device_name: Smart Meter
          device_img: ""
          create_time: 1722369608
          status: "1"
          bind_site_status: ""
          wireless_type: ""
      photovoltaic_to_grid_power: "0"
      grid_to_home_power: "0"
      grid_status: 0
      has_history_data: true
    is_downgrade: false
    error_code: 0
    smart_plug_info: null
    feature_switch:
      multi_pv: true
      smart_plug: true
    other_loads_power: "365"

but this time my HA is updating like it should 👍 

![image](https://github.com/user-attachments/assets/4b0ca685-7701-4219-913b-cdd1f1ddea06)

**Reply by kimi1983** [reply]
and now even SB Datenzeit is correct, after some time....strange.

Anyway, thank you for this great integration :)

**Reply by kimi1983** [reply]
andy ma latest output, looking fine 👍 

sensor.system_mini_pv_ertrag_gesamt:
  system_info:
    home_info:
      home_name: Home
      home_img: ""
      charging_power: "0.00"
      power_unit: W
    solar_list: []
    pps_info:
      pps_list: []
      total_charging_power: "0.00"
      power_unit: W
      total_battery_power: "0.00"
      updated_time: ""
      pps_status: 0
    statistics:
      - type: "1"
        total: "0.83"
        unit: kWh
      - type: "2"
        total: "0.83"
        unit: kg
      - type: "3"
        total: "0.17"
        unit: €
    topology_type: "1"
    solarbank_info:
      solarbank_list:
        - device_pn: A17C1
          device_sn: APCGQ80E22200075
          device_name: Solarbank 2 E1600 Pro
          device_img: >-
            https://public-aiot-fra-prod.s3.dualstack.eu-central-1.amazonaws.com/anker-power/public/product/2024/05/24/iot-admin/5iJoq1dk63i47HuR/picl_A17C1_normal%281%29.png
          battery_power: "21"
          bind_site_status: ""
          charging_power: "337"
          power_unit: W
          charging_status: "3"
          status: "1"
          wireless_type: "1"
          main_version: ""
          photovoltaic_power: "884"
          output_power: "337"
          create_time: 1722373844
          set_load_power: ""
          sub_package_num: 1
          output_cutoff_data: 10
          is_display: true
      total_charging_power: "547"
      power_unit: W
      charging_status: "0"
      total_battery_power: "0.21"
      updated_time: "2024-07-31 08:53:23"
      total_photovoltaic_power: "884"
      total_output_power: "337.00"
      display_set_power: false
      battery_discharge_power: "0"
      ac_power: "0"
      to_home_load: "337"
      is_display_data: true
      solar_power_1: "222"
      solar_power_2: "220"
      solar_power_3: "223"
      solar_power_4: "219"
    retain_load: 0W
    scene_mode: 1
    home_load_power: "337"
    updated_time: 01-01-0001 00:00:00
    power_site_type: 5
    site_id: c8eccfab-f02a-4c09-bec7-76adc9a3776f
    powerpanel_list: []
    grid_info:
      grid_list:
        - device_pn: A17X7
          device_sn: AUJS020E17412691
          device_name: Smart Meter
          device_img: ""
          create_time: 1722369608
          status: "1"
          bind_site_status: ""
          wireless_type: ""
      photovoltaic_to_grid_power: "0"
      grid_to_home_power: "0"
      grid_status: 0
      has_history_data: true
    is_downgrade: false
    error_code: 0
    smart_plug_info: null
    feature_switch: null
    other_loads_power: "337"

**Reply by thomluther** [reply]
Maybe it took some time to synchronize the time.
Anyway, when the integration receives a response with such an invalid timestamp:
updated_time: "1970-01-01 00:00:00"
It will be replaced with the actual time of your Home Assistant OS. So if you still see a gap of 2 hours from time to time, it means that either the Cloud, the solarbank or your HA timezone is not in sync...

**Comment by asemev** [comment]
Hello, 

I opened a new issue: [https://github.com/thomluther/hacs-anker-solix/issues/100]

I asked the discord forum as well and they said that the 4 entities's state class needs to be changed to total_increasing so that the energy dashboard works correctly.

Here is what Karwosts wrote:

karwosts — Today at 14:19
Ok those energy sensors seem like they should be state_class: total_increasing and not total
as they are, when they reset to 0 every night, that's interpreted as a big negative change instead of an accumulation reset.

Can you please have a look? 
many thanks.

**Reply by thomluther** [reply]
That is correct, this will be fixed in the final version.
But as written in the issue, I do NOT recommend to use the integration entities directly in the dashboard.
The dashboard integration is discussed [here ](https://github.com/thomluther/hacs-anker-solix/discussions/16)in detail
